repository/scripts/items/FoodTypes.js

```diff
@@ -300,80 +298,52 @@
 	        healText : null
 	    }	    			    		    		    	   
 };
- 
-var ItemListener = Java.extend(Java.type('org.virtue.engine.script.listeners.EventListener'), {
-	invoke : function (event, objTypeId, args) {
-		var player = args.player;
-		var item = args.item;
-		var slot = args.slot;
-		
-		startFood(player, item, slot);
-	}
-});
 
-/* Listen to the item ids specified */
-var listen = function(scriptManager) {
-	var ids = [ 1973, 319, 2309, 13433, 2140, 2142, 315, 325, 347, 355, 333, 339, 351, 
-	    		329, 361, 365, 379, 373, 7946, 385, 397, 15266, 391, 15272, 1891, 1897, 2289, 
-	    		2293, 2297, 2301, 7178, 6705, 7056, 7060, 7208, 7218, 21521, 19948, 19949, 26313, 6685, 23351, 28191, 28227 ];
-	var itemListener = new ItemListener();
-	for (var i in ids) {
-		scriptManager.registerListener(EventType.OPHELD1, ids[i], itemListener);
-	}
-}
+   return {
+		init : init
+	};
+
 
-function startFood (player, item, slot) {
-	if (api.isPaused(player)) {
-		return false;
+	function init (scriptManager) {
+	var ids = [];
+	for (var i in Types) {
+	ids.push(Types[i].itemID);
 	}
-	var food = forFood(player, item);
+    for (var i in ids) {
+	scriptManager.bind(EventType.OPHELD1, ids[i], function (ctx) {
+	var food = forFood(ctx.player, ctx.item);
 	if (food == null) {
 		return;
 	}
 	var delay = 2;
-	api.freezeEntity(player, delay+1);
+	ENGINE.freezeEntity(ctx.player, delay+1);
 	if (food.healText != null) {
-		api.sendFilterMessage(player, food.healText);		
+		chat.sendMessage(ctx.player, food.healText);		
 	} else {
-		api.sendFilterMessage(player, "You eat the " + api.getItemName(food.itemID) + ".");
+		chat.sendMessage(ctx.player, "You eat the " + ENGINE.getItemName(food.itemID) + ".");
```

Comment:
There's a config method you could use here (`config.objName()` instead of `ENGINE.getItemName()`)